### PR TITLE
Readme: Add architecture diagram

### PR DESCRIPTION
**Pull request by ebugden**
If approved, before merging:
- [x] Change image path to master in the main repo

Signed-off-by: Erica Bugden <erica.bugden@gmail.com>

### CODE REVIEW

**Comment by ebugden** [comment]
Should this "Architecture" section go above or below the "Related code" section?

File: README.md [hunk 1]
```diff
@@ -225,3 +225,9 @@ This trace viewer depends on code from several other repos. Sometimes resolving
 | [Trace Server Protocol (TSP)](https://github.com/theia-ide/trace-server-protocol) | Protocol used by the trace viewer to communicate with the trace server. | [label:"trace server protocol"](https://github.com/theia-ide/theia-trace-extension/labels/trace%20server%20protocol) |
 | [Client-side Trace Server Protocol implementation](https://github.com/theia-ide/tsp-typescript-client) | A client-side implementation of the Trace Server Protocol. Allows the trace viewer to communicate with the server. |  |   |
 | [Timeline Chart](https://github.com/theia-ide/timeline-chart) | Implements the Gantt charts used in this trace viewer.  | [label:timeline-chart](https://github.com/theia-ide/theia-trace-extension/labels/timeline-chart) |   |
+
+## Architecture
+The diagram below shows the main parts of the Trace Viewer's front end (left side) and back end (right side) as well as how they exchange information.
+
+![Trace viewer architecture diagram](https://raw.githubusercontent.com/ebugden/theia-trace-extension/architecture-diagram/doc/images/theia-trace-viewer-architecture.PNG)
```
Comment (bhufmann) [inline README.md:232]: Make sure it points to the one in `theia-ide`.

**Review by bhufmann — CHANGES_REQUESTED** [review]
Looks good to me. Please update the link to the image and rebase to the latest master.

**Review by bhufmann — APPROVED** [review]

**Comment by bhufmann** [comment]
@ebugden could you please rebase this PR to latest master?

**Review by PatrickTasse — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (bhufmann) [reply README.md:233]: @ebugden this needs to be still updated to point to the file in this repo.

**Review by bhufmann — CHANGES_REQUESTED** [review]
Please update this PR with right link to the file. I did a rebase on GitHub, but  if you could just update the PR and make sure that it's rebased on master, then you can remove my merge commit. I prefer actually not to have the merge commit.

**Comment by ebugden** [comment]
Update:
* Change image path to pull from main repo
* Rebase on master

**Review by bhufmann — APPROVED** [review]
Thanks for the updates. LGTM.

**Review by PatrickTasse — APPROVED** [review]

**Pull request merged**